server/client.go

```diff
@@ -797,12 +797,6 @@ func (c *client) processMsg(msg []byte) {
 	// Snapshot server.
 	srv := c.srv
 
-	// Create cache subs map if needed.
-	if c.cache.results == nil && srv != nil {
```

Comment:
The if is completly in line 829. srv can never be nil after line 819 and event if it is we will return without processing the message, so we don't even need to initialize the cache.